dbus/subscription.go

```diff
@@ -227,10 +251,13 @@ func (c *Conn) shouldIgnore(path dbus.ObjectPath) bool {
 }
 
 func (c *Conn) updateIgnore(path dbus.ObjectPath, info map[string]interface{}) {
-	c.cleanIgnore()
```

Comment:
Why move this out of here?

(seems fine to me either way, but if anyone other than `sendSubStateUpdate` starts using the ignore functions, then they'll have to remember to call `cleanIgnore` themselves)

Reply:
Because `updateIgnore` is only called after we a proper and non-ignored `info`, while `cleanIgnore` is generic bookkeeping that doesn't need any of it. I was just reading the code, and I felt it was out of place, so I turned it into a non-conditional defer.

You are right about the multiple callers. This code may need some more intrusive re-architecting once we get there, the whole ignore thing is quite hackish.

Reply:
Makes sense—sounds good.